LEGO1/omni/include/mxminiaudio.h

```diff
@@ -0,0 +1,64 @@
+#ifndef MXMINIAUDIO_H
+#define MXMINIAUDIO_H
+
+#include "mxtypes.h"
+
+#include <assert.h>
+#include <miniaudio.h>
+#include <utility>
+
+template <typename T>
+class MxMiniaudio {
+public:
+	MxMiniaudio() : m_initialized(false) {}
+
+	template <typename Fn, typename... Args>
+	ma_result Init(Fn p_init, Args&&... p_args)
+	{
+		assert(!m_initialized);
+
+		ma_result result = p_init(std::forward<Args>(p_args)..., &m_object);
+		if (result == MA_SUCCESS) {
+			m_initialized = true;
+		}
+
+		return result;
+	}
+
+	template <typename Fn>
+	void Destroy(Fn p_uninit)
+	{
+		if (m_initialized) {
+			p_uninit(&m_object);
+			m_initialized = false;
+		}
+	}
+
+	T* operator->()
+	{
+		assert(m_initialized);
+		if (m_initialized) {
+			return &m_object;
+		}
+
+		return nullptr;
+	}
```

Comment:
Do you still need this operator?
A normal `->` would return a `T` reference, but this again allows abuse. So perhaps better to remove it?
```suggestion
	T& operator->()
	{
		assert(m_initialized);
		return m_object;
	}
```

Reply:
We need it in a few places:

![image](https://github.com/user-attachments/assets/94700b47-9c30-43b2-af48-0df9f9e2345a)

Reply:
Awesome! Merge this sh*t then :)